docx/bookmark.py

```diff
@@ -65,3 +65,19 @@ def __init__(self, part):
     def iter_start_end_pairs(cls, part):
         """Generate each (bookmarkStart, bookmarkEnd) in *part*."""
         return cls(part)._iter_start_end_pairs()
+
+    def _iter_start_end_pairs(self):
+        """Generate each (bookmarkStart, bookmarkEnd) in this part."""
+        raise NotImplementedError
+
+    def _iter_starts(self):
+        """Generate (idx, bookmarkStart) elements in story.
+
+        The *idx* value indicates the location of the bookmarkStart element
+        among all the bookmarkStart and bookmarkEnd elements in the story.
+        """
+        raise NotImplementedError
+
+    @lazyproperty
+    def _all_starts_and_ends(self):
+        return self._part.element.xpath('//w:bookmarkStart|//w:bookmarkEnd')
```

Comment:
I hope this is acceptable; I didn't know how to proceed, as I wasn't allowed to test 'two'  features.  Therefore I started at the `_all_starts_and_ends` method and worked my way backwards.